### PR TITLE
UI req 990 add dropdown for item service point

### PR DESCRIPTION
**Pull request by rgupta-nla**
## Purpose
https://issues.folio.org/browse/UIREQ-989
To populate the drop down the (ServicePoint) code will need to be used, this allows the already existing service point code to be leveraged.

User must have the view request permission. Users: View requests, this should also limit them to their assigned service point/s. (This may need to be updated to suit general open source) This does match the generic permission in snapshot.

## Approach
The request list is filtered in the backend based on the item selected in Primary Service Point name dropdown in the ui.

image

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -221,6 +221,7 @@ export const requestFilterTypes = {
   REQUEST_STATUS: 'requestStatus',
   REQUEST_LEVELS: 'requestLevels',
   PICKUP_SERVICE_POINT: 'pickupServicePoints',
+  EffLocation_SERVICE_POINT: 'effectiveLocationPrimaryServicePoint',
```
Comment (fcrimmins) [inline src/constants.js:224]: Should this be an all-caps constant name to match the other types?

**Review by fcrimmins — APPROVED** [review]

File: src/components/RequestsFilters/EffectiveLocationServicePointFilter/EffectiveLocationServicePointFilter.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    }));
+  }, [servicePoints]);
+
+  return (
+    <Accordion
+      data-testid="effectiveLocationServicePointAccordion"
+      displayClearButton={!isEmpty(activeValues)}
+      id={name}
+      header={FilterAccordionHeader}
+      label="Effective Location Primary Service Point Name"
```
Comment (Dmitriy-Litvinenko) [inline src/components/RequestsFilters/EffectiveLocationServicePointFilter/EffectiveLocationServicePointFilter.js:45]: Could you please move label text to translation?
Could you please clarify label (current text not consistency)?

File: src/components/RequestsFilters/EffectiveLocationServicePointFilter/EffectiveLocationServicePointFilter.test.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import {
+  render,
+  screen,
+} from '@folio/jest-config-stripes/testing-library/react';
+import userEvent from '@folio/jest-config-stripes/testing-library/user-event';
+
+import '__mock__';
```
Comment (Dmitriy-Litvinenko) [inline src/components/RequestsFilters/EffectiveLocationServicePointFilter/EffectiveLocationServicePointFilter.test.js:7]: Could you  please remove?

File: translations/ui-requests/en.json [hunk 4]
```diff
@@ -161,6 +161,7 @@
   "item.status.orderClosed": "Order closed",
   "item.status.restricted": "Restricted",
   "title.information": "Title information",
+  "requests.itemEffectiveLocationPrimaryServicePointName": "Effective Location Primary Service Point Name",
```
Comment (Dmitriy-Litvinenko) [inline translations/ui-requests/en.json:164]: Could you please clarify text (current text not consistency)?

**Review by Dmitriy-Litvinenko — CHANGES_REQUESTED** [review]

**Comment by Dmitriy-Litvinenko** [comment]
Could you please resolve conflict and fix feedback?

**Comment by Dmitriy-Litvinenko** [comment]
We close current pull request. Pull request long time without any updates, need rebase and etc. If needed can be reopened.

**Pull request closed without merging**